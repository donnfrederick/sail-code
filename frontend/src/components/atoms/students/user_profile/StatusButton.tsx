import * as React from 'react'

import Button from 'components/atoms/students/Button'
import FavoriteButton from 'components/atoms/students/FavoriteButton'
import ConfirmAddFavorite from 'components/organisms/students/modal_contents/confirm_add_favorite'
import ConfirmRemoveFavorite from 'components/organisms/students/modal_contents/confirm_remove_favorite'
import ConfirmUnblock from 'components/organisms/students/modal_contents/confirm_unblock'

interface Props {
  isBlocked: boolean
  isFavorite: boolean
  userId: number
  open(): void
  setContents(contents: JSX.Element): void
}

export default ({
  isBlocked,
  isFavorite,
  userId,
  setContents,
  open
}: Props) => {
  return isBlocked ? (
    <Button
      type="red"
      text="Blocked"
      fontSize={28}
      width={210}
      height={56}
      onClick={() => {
        setContents(<ConfirmUnblock blockedUserId={userId} />)
        open()
      }}
    />
  ) : (
    <FavoriteButton
      isFavorite={isFavorite}
      onClick={() => {
        setContents(
          isFavorite ? (
            <ConfirmRemoveFavorite favoritedUserId={userId} />
          ) : (
            <ConfirmAddFavorite addFavoritedUserId={userId} />
          )
        )
        open()
      }}
    />
  )
}
