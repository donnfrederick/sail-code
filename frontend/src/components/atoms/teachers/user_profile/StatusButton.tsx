import * as React from 'react'

import Button from 'components/atoms/teachers/Button'
import FavoriteButton from 'components/atoms/teachers/FavoriteButton'
import ConfirmAddFavorite from 'components/organisms/teachers/modal_contents/confirm_add_favorite'
import ConfirmRemoveFavorite from 'components/organisms/teachers/modal_contents/confirm_remove_favorite'
import ConfirmUnblock from 'components/organisms/teachers/modal_contents/confirm_unblock'

interface Props {
  isBlocked: boolean
  isFavorite: boolean
  userId: number
  open(): void
  setContents(contents: JSX.Element): void
}

export default ({ isBlocked, userId, setContents, isFavorite, open }: Props) =>
  isBlocked ? (
    <Button
      type="red"
      text="ブロック中"
      fontSize={28}
      width={210}
      height={56}
      onClick={() => {
        setContents(<ConfirmUnblock blockedUserId={userId} />)
        open()
      }}
    />
  ) : (
    <FavoriteButton
      isFavorite={isFavorite}
      onClick={() => {
        setContents(
          isFavorite ? (
            <ConfirmRemoveFavorite favoritedUserId={userId} />
          ) : (
            <ConfirmAddFavorite addFavoritedUserId={userId} />
          )
        )
        open()
      }}
    />
  )
