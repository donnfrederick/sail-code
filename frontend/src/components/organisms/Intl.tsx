import React, {
  PropsWithChildren,
  Dispatch,
  SetStateAction,
  useContext,
  useState
} from 'react'
import { IntlProvider } from 'react-intl'
import { messages, browserLocale, Locale } from 'utils/intl'

export type LocaleState = {
  locale: Locale
}

const initialState: LocaleState = {
  locale: browserLocale
}

const IntlStateContext = React.createContext<LocaleState>(initialState)

const SetLocaleStateContext = React.createContext<
  Dispatch<SetStateAction<LocaleState>>
>(() => {
  // empty block
})

export function useLocaleState() {
  return useContext(IntlStateContext)
}
export function useSetLocaleState() {
  return useContext(SetLocaleStateContext)
}

export function LocaleStateProvider(props: {
  initialState?: LocaleState
  children: React.ReactNode
}) {
  const [state, setState] = useState<LocaleState>(
    props.initialState || initialState
  )
  return (
    <IntlStateContext.Provider value={state}>
      <SetLocaleStateContext.Provider value={setState}>
        {props.children}
      </SetLocaleStateContext.Provider>
    </IntlStateContext.Provider>
  )
}

export const Intl = (props: PropsWithChildren<{}>) => {
  const localeState = useLocaleState()

  return (
    <LocaleStateProvider>
      <IntlProvider
        messages={messages[localeState.locale]}
        locale={localeState.locale}
        defaultLocale="en"
      >
        {props.children}
      </IntlProvider>
    </LocaleStateProvider>
  )
}
