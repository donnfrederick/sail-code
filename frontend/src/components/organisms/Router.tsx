import * as Sentry from '@sentry/browser'
import ErrorPage from 'components/pages/Error'
import { createBrowserHistory } from 'history'
import * as React from 'react'
import { Provider, Store } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { compose, lifecycle, withState } from 'recompose'
import createStore from 'store/createStore'
import resetViewport from 'utils/resetViewport'
import userAgent from 'utils/userAgent'

declare const gtag: any

export const history = createBrowserHistory()
export const store = createStore(history) as Store<any>

history.listen(() => {
  window.scrollTo(0, 0)
  gtag('config', 'UA-105776977-2', { page_path: history.location.pathname })
  userAgent()
  resetViewport()
})

interface Props {
  children: JSX.Element
  errors: Error | null
  setErrors: (state: any) => any
}

const isProd = process.env.NODE_ENV === 'production'

const sentry = compose(
  withState('errors', 'setErrors', null),
  isProd
    ? lifecycle<Props, {}, {}>({
        componentDidCatch(error, errorInfo) {
          this.props.setErrors(error)
          Sentry.withScope(scope => {
            Object.keys(errorInfo).forEach(key => {
              scope.setExtra(key, errorInfo[key])
            })
            Sentry.captureException(error)
          })
        }
      })
    : lifecycle<Props, {}, {}>({
        componentDidCatch(error) {
          this.props.setErrors(error)
        }
      })
)

export default sentry((props: Props) => {
  const { children, errors } = props

  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        {errors ? <ErrorPage /> : children}
      </ConnectedRouter>
    </Provider>
  )
})
