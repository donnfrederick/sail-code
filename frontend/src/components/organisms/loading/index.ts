import Loading from 'components/organisms/loading/Loading'
import { connect } from 'react-redux'
import { State as RootState } from 'reducers'

// tslint:disable-next-line
interface OuterProps {}

interface Props {
  isFetchingConversation: boolean
  isFetchingLocations: boolean
  isFetchingNotification: boolean
  isFetchingOrganizations: boolean
  isFetchingStudentsMe: boolean
  isFetchingTeachersMe: boolean
}

const mapStateToProps = (state: RootState, props: OuterProps): Props => {
  return {
    isFetchingConversation: state.rootReducer.conversation.fetching > 0,
    isFetchingLocations: state.rootReducer.locations.isFetching,
    isFetchingNotification: state.rootReducer.notification.isFetching,
    isFetchingOrganizations: state.rootReducer.organizations.isFetching,
    isFetchingStudentsMe: state.rootReducer.students.isFetching,
    isFetchingTeachersMe: state.rootReducer.teachers.isFetching
  }
}

const enhancer: any = connect(
  mapStateToProps,
  {}
)

export default enhancer(Loading)
