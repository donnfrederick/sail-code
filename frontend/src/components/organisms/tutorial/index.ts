import Tutorial from 'components/organisms/tutorial/Tutorial'
import { connect } from 'react-redux'
import { State as RootState } from 'reducers'
import * as TutorialActions from 'reducers/tutorial'

// tslint:disable-next-line
interface OuterProps {}

interface Props {
  shouldShow: boolean
  showConfirmation: boolean
}

const mapStateToProps = (state: RootState, props: OuterProps): Props => {
  return {
    shouldShow: state.rootReducer.tutorial.shouldShow,
    showConfirmation: state.rootReducer.teachers.showConfirmation
  }
}

const enhancer: any = connect(
  mapStateToProps,
  {
    hide: TutorialActions.hide,
    show: TutorialActions.show
  }
)

export default enhancer(Tutorial)
