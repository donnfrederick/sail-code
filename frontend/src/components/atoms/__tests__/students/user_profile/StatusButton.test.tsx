import StatusButton from 'components/atoms/students/user_profile/StatusButton'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Intl } from 'components/organisms/Intl'

test('render <StatusButton />', () => {
  const setContentsFn = (contents: JSX.Element) => {}
  const tree = renderer
    .create(
      <Intl>
        <StatusButton
          userId={1}
          isBlocked={true}
          isFavorite={true}
          open={() => {}}
          setContents={setContentsFn}
        />
      </Intl>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
