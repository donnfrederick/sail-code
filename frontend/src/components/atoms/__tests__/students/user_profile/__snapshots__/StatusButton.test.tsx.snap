// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <StatusButton /> 1`] = `
<button
  className="sc-bdVaJa btQiBq"
  data-active={true}
  fontSize={28}
  height={56}
  onClick={[Function]}
  type="red"
  width={210}
>
  Blocked
</button>
`;
